src/components/calendar/CalendarRegistrator.tsx

```diff
@@ -0,0 +1,235 @@
+import React from 'react';
+import { View, KeyboardAvoidingView, TouchableWithoutFeedback, Keyboard, Platform  } from 'react-native';
+import { IconButton, Avatar, useTheme } from 'react-native-paper';
+
+import { DatePicker } from '../DatePicker';
+import { TimePicker, getCurrentTimeRounded } from '../TimePicker';
+import { ChoiceBasedTextInput } from '../ChoiceBasedTextInput';
+
+import { useTranslation } from 'language/LanguageProvider';
+import Storage from 'storage';
+import { CalendarEntry } from 'storage/types';
+import { RouteProp } from '@react-navigation/core';
+import { CalendarStackParamList } from '../CalendarScreen';
+import { StackNavigationProp } from '@react-navigation/stack';
+
+/**
+ * Returns the date in ISO format. Normal programming languages use
+ * something like date.strftime("%Y-%m-%d").
+ *
+ * @returns A ISO-formatted date
+ */
+const ISODate = (date: Date): string => {
+  const year = date.getFullYear();
+  let month = '' + (date.getMonth() + 1);
+  let day = '' + date.getDate();
+
+  if (month.length < 2) {
+    month = '0' + month;
+  }
+  if (day.length < 2) {
+    day = '0' + day;
+  }
+
+  return [year, month, day].join('-');
+}
+
+const ISOTime = (date: Date): string => {
+  let hour = '' + date.getHours();
+  let minute = '' + date.getMinutes();
+
+  if (hour.length < 2) {
+    hour = '0' + hour;
+  }
+  if (minute.length < 2) {
+    minute = '0' + minute;
+  }
+
+  return hour + ':' + minute;
+}
+
+type CalendarRegistratorRouteProp = RouteProp<
+  CalendarStackParamList,
+  'CalendarRegistration'
+>;
+
+type CalendarRegistratorNavigationProp = StackNavigationProp<
+  CalendarStackParamList,
+  'CalendarRegistration'
+>;
+
+type Props = {
+  route: CalendarRegistratorRouteProp,
+  navigation: CalendarRegistratorNavigationProp,
+};
+
+export const CalendarRegistrator = ({ route, navigation }: Props) => {
+  const lang = useTranslation();
+  const { iconSizes, colors } = useTheme();
+
+  const steps = 2;
```

Comment:
```suggestion
  const steps = 4;
```

I think the Customer wanted that you colud add a planning activity every quarter